### PR TITLE
Fix indices in modeler

### PR DESCRIPTION
**Pull request by guilpier-code**
This PR for a discussion on how we should handle the **modeler** regarding the **short term storage** functionality.

### CODE REVIEW

File: src/solver/optimisation/opt_gestion_des_bornes_cas_lineaire.cpp [hunk 1]
```diff
@@ -173,9 +173,9 @@ static void setBoundsForShortTermStorage(PROBLEME_HEBDO* problemeHebdo,
           = problemeHebdo->CorrespondanceVarNativesVarOptim[pdtJour];
         for (int areaIndex = 0; areaIndex < problemeHebdo->NombreDePays; areaIndex++)
         {
-            int areaWideIndex = 0;
             for (const auto& storage : (*problemeHebdo->ShortTermStorage)[areaIndex])
             {
+                int storageIndex = 0;
```
Comment (flomnes) [inline src/solver/optimisation/opt_gestion_des_bornes_cas_lineaire.cpp]: It seems that storageIndex is always 0, right ?

File: src/solver/simulation/sim_calcul_economique.cpp [hunk 2]
```diff
@@ -43,14 +43,15 @@ static void importShortTermStorages(
...
   std::vector<::ShortTermStorage::AREA_INPUT>& ShortTermStorageOut)
 {
-    int STindex = 0;
+    int globalIndex = 0;
     for (uint areaIndex = 0; areaIndex != areas.size(); areaIndex++)
     {
         ShortTermStorageOut[areaIndex].resize(areas[areaIndex]->shortTermStorage.count());
         for (auto st : areas[areaIndex]->shortTermStorage.storagesByIndex)
         {
+            int STindex = 0;
```
Comment (flomnes) [inline src/solver/simulation/sim_calcul_economique.cpp]: Same remark, STindex is always 0.

**Review by flomnes — CHANGES_REQUESTED** [review]

File: src/solver/optimisation/opt_gestion_des_bornes_cas_lineaire.cpp (hunk 1 above)
Comment (guilpier-code) [reply src/solver/optimisation/opt_gestion_des_bornes_cas_lineaire.cpp]: oops, corrected.

File: src/solver/simulation/sim_calcul_economique.cpp (hunk 2 above)
Comment (guilpier-code) [reply src/solver/simulation/sim_calcul_economique.cpp]: oops, corrected.

File: src/solver/optimisation/opt_gestion_des_bornes_cas_lineaire.cpp
Comment (flomnes) [inline src/solver/optimisation/opt_gestion_des_bornes_cas_lineaire.cpp:1]: Only renaming here

File: src/solver/simulation/sim_calcul_economique.cpp
Comment (flomnes) [inline src/solver/simulation/sim_calcul_economique.cpp:1]: In this function, you set `toInsert.globalIndex` to the global index, not the local index.

I agree with this change. Strange that we never found it, it shows that our tests are insufficient.

**Review by flomnes — APPROVED** [review]

File: src/solver/simulation/sim_calcul_economique.cpp
Comment (flomnes) [reply src/solver/simulation/sim_calcul_economique.cpp:1]: Also, consider renaming `STindex` to `storageIndex` for consistency.

**Review by payetvin — APPROVED** [review]

**Pull request merged**